speech/recognize.js

```diff
@@ -11,106 +11,140 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-'use strict';
+/**
+ * This application demonstrates how to perform basic recognize operations with
+ * with the Google Cloud Speech API.
+ *
+ * For more information, see the README.md under /speech and the documentation
+ * at https://cloud.google.com/speech/docs.
+ */
 
-// [START app]
-// [START import_libraries]
-var google = require('googleapis');
-var async = require('async');
-var fs = require('fs');
+'use strict';
 
-// Get a reference to the speech service
-var speech = google.speech('v1beta1').speech;
-// [END import_libraries]
+const fs = require('fs');
+const record = require('node-record-lpcm16');
+const speech = require('@google-cloud/speech')();
 
-// [START authenticating]
-function getAuthClient (callback) {
-  // Acquire credentials
-  google.auth.getApplicationDefault(function (err, authClient) {
+// [START speech_sync_recognize]
+function syncRecognize (filename, callback) {
+  // Detect speech in the audio file, e.g. "./resources/audio.raw"
+  speech.recognize(filename, {
+    encoding: 'LINEAR16',
+    sampleRate: 16000
+  }, (err, results) => {
```

Comment:
With this PR, I decided to start using the ES2015 features of JavaScript that are supported in the latest 4.5.0 LTS release of Node.js. See https://github.com/nodejs/LTS and http://node.green/

With ES2015, one would use arrow functions (`(a, b) => { ...`) as much as possible. You'll notice in this PR I also started using the `const` keyword. I also added an `engines` field to `package.json`.